webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -0,0 +1,287 @@
+    updateSettings({ row: undefined });
+  }, [ updateSettings ]);
+
+  const handleStateFilterApply = useCallback((states: string[]) => {
+    updateSettings({
+      row: undefined,
+      state: states.length !== 0 ? states as CheckpointState[] : undefined,
+    });
+  }, [ updateSettings ]);
+
+  const handleStateFilterReset = useCallback(() => {
+    updateSettings({ row: undefined, state: undefined });
+  }, [ updateSettings ]);
+
+  const stateFilterDropdown = useCallback((filterProps: FilterDropdownProps) => (
+    <TableFilterDropdown
+      {...filterProps}
+      multiple
+      values={settings.state}
+      onFilter={handleStateFilterApply}
+      onReset={handleStateFilterReset}
+    />
+  ), [ handleStateFilterApply, handleStateFilterReset, settings.state ]);
+
+  const handleRegisterCheckpoint = useCallback((checkpoint: CoreApiGenericCheckpoint) => {
+    openModalCheckpointRegister({ checkpointUuid: checkpoint.uuid });
+  }, [ openModalCheckpointRegister ]);
+
+  const dropDownOnTrigger = useCallback((checkpoint: CoreApiGenericCheckpoint) => {
+    return { [CheckpointAction.Register]: () => handleRegisterCheckpoint(checkpoint) };
+  }, [ handleRegisterCheckpoint ]);
+
+  const columns = useMemo(() => {
+    const actionRenderer = (_: string, record: CoreApiGenericCheckpoint): React.ReactNode => (
+      <ActionDropdown<CheckpointAction>
+        actionOrder={[
+          CheckpointAction.Register,
+        ]}
+        id={record.uuid}
+        kind="checkpoint"
+        onError={handleError}
+        onTrigger={dropDownOnTrigger(record)}
+      />
+    );
+
+    const checkpointRenderer = (_: string, record: CoreApiGenericCheckpoint): React.ReactNode => {
+      return (
+        <CheckpointModalTrigger
+          checkpoint={record}
+          experiment={experiment}
+          title={`Checkpoint ${record.uuid}`}
+        />
+      );
+    };
+
+    const newColumns = [ ...defaultColumns ].map((column) => {
+      column.sortOrder = null;
+      if (column.key === 'checkpoint') {
+        column.render = checkpointRenderer;
+      } else if (column.key === V1GetExperimentCheckpointsRequestSortBy.STATE) {
+        column.filterDropdown = stateFilterDropdown;
+        column.isFiltered = (settings) => !!(settings as Settings).state;
+        column.filters = ([ 'ACTIVE', 'COMPLETED', 'DELETED', 'ERROR' ] as CheckpointState[])
+          .map((value) => ({
+            text: <Badge state={value} type={BadgeType.State} />,
+            value,
+          }));
+      } else if (column.key === 'actions') {
+        column.render = actionRenderer;
+      }
+      if (column.key === settings.sortKey) {
+        column.sortOrder = settings.sortDesc ? 'descend' : 'ascend';
+      }
+      return column;
+    });
+
+    return newColumns;
+  }, [ dropDownOnTrigger, experiment, settings.sortDesc, settings.sortKey, stateFilterDropdown ]);
+
+  const handleTableChange = useCallback((tablePagination, tableFilters, tableSorter) => {
```

Comment:
question: why do we need to supply our own `handleTableChange` instead of just using the one already in `InteractiveTable`?

Reply:
Again, copied from `ExperimentTrials`. Did some testing and you're right, it shouldn't be necessary.